presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/mysql/MySqlFunctionNamespaceManager.java

```diff
@@ -0,0 +1,337 @@
+
+            checkExists(functions, functionName, parameterTypes);
+
+            if (!parameterTypes.isPresent()) {
+                transactionDao.setDeleted(functionName.getCatalogName(), functionName.getSchemaName(), functionName.getObjectName());
+            }
+            else {
+                SqlInvokedFunction latest = getOnlyElement(functions);
+                checkState(latest.hasVersion(), "Function version missing: %s", latest.getFunctionId());
+                transactionDao.setDeletionStatus(hash(latest.getFunctionId()), latest.getFunctionId(), getLongVersion(latest), true);
+            }
+        });
+
+        refreshFunctionsCache(functionName);
+    }
+
+    private List<SqlInvokedFunction> getSqlFunctions(FunctionNamespaceDao functionNamespaceDao, QualifiedObjectName functionName, Optional<List<TypeSignature>> parameterTypes)
+    {
+        List<SqlInvokedFunctionRecord> records = new ArrayList<>();
+        if (parameterTypes.isPresent()) {
+            SqlFunctionId functionId = new SqlFunctionId(functionName, parameterTypes.get());
+            functionNamespaceDao.getLatestRecordForUpdate(hash(functionId), functionId).ifPresent(records::add);
+        }
+        else {
+            records = functionNamespaceDao.getLatestRecordsForUpdate(functionName.getCatalogName(), functionName.getSchemaName(), functionName.getObjectName());
+        }
+        return records.stream()
+                .filter(record -> !record.isDeleted())
+                .map(SqlInvokedFunctionRecord::getFunction)
+                .collect(toImmutableList());
+    }
+
+    private void insertSqlInvokedFunction(FunctionNamespaceDao functionNamespaceDao, SqlInvokedFunction function, long version)
+    {
+        QualifiedObjectName functionName = function.getFunctionId().getFunctionName();
+        functionNamespaceDao.insertFunction(
+                hash(function.getFunctionId()),
+                function.getFunctionId(),
+                version,
+                functionName.getCatalogName(),
+                functionName.getSchemaName(),
+                functionName.getObjectName(),
+                function.getParameters(),
+                function.getSignature().getReturnType(),
+                function.getDescription(),
+                function.getRoutineCharacteristics(),
+                function.getBody());
+    }
+
+    private static long getLongVersion(SqlFunctionHandle functionHandle)
+    {
+        return parseLong(functionHandle.getVersion());
+    }
+
+    private static long getLongVersion(SqlInvokedFunction function)
+    {
+        return parseLong(function.getRequiredVersion());
+    }
+
+    private static void checkFieldLength(String fieldName, String field, int maxLength)
+    {
+        if (field.length() > maxLength) {
+            throw new PrestoException(GENERIC_USER_ERROR, format("%s exceeds max length of %s: %s", fieldName, maxLength, field));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: maxLength (declared at line 305). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 308. Consider formatting this data with &#39;%d&#39; instead.